### DISCUSSION TITLE
Using File Uploads on Vapor

### DISCUSSION THREAD

**Discussion by shanerbaner82** [livewire/livewire#1161] [4. General]
[PR](https://github.com/livewire/livewire/pull/1160)

Is anyone else running into this? Laravel Vapor has [Reserved Keys](https://docs.vapor.build/1.0/projects/environments.html#environment-variables) that Livewire is statically looking for for S3 File Uploads?

**Comment by shanerbaner82** [comment]
Forgot to add "[storage](https://docs.vapor.build/1.0/resources/storage.html#introduction)" to vapor.yml

**Comment by calebporzio** [comment]
Hey @shanerbaner82, is this a problem that other Vapor users will hit? Meaning is it in the happy-path, or is this something obscure you've done and other people on the edge will discover.

Thanks!

**Reply by shanerbaner82** [reply]
Not entirely sure how obscure it is @calebporzio, but sometimes I have used a different bucket for S3, as long as you use the default bucket for Vapor everything is good. This might be a nice feature to add though if you want to use a different bucket than the default Vapor bucket. Or, I guess I should say the bucket you specify in Vapor storage, Vapor has to know which bucket to use in the current implementation..
